packages/js/src/core/fetch-token.ts

```diff
@@ -21,15 +22,21 @@ export type FetchTokenByRefreshTokenParameters = {
   scopes?: string[];
 };
 
-type TokenSnakeCaseResponse = {
+type SnakeCaseCodeTokenResponse = {
   access_token: string;
   refresh_token: string;
-  id_token?: string;
+  id_token: string;
   scope: string;
   expires_in: number;
 };
 
-export type RefreshTokenTokenResponse = KeysToCamelCase<TokenSnakeCaseResponse>;
+type SnakeCaseRefreshTokenTokenResponse = Except<SnakeCaseCodeTokenResponse, 'id_token'> & {
+  id_token?: string;
+};
```

Comment:
A new type `RefreshTokenTokenResponse` would be better?
This line of code seems hacked.

Reply:
`RefreshTokenTokenResponse` will change in the future.

Reply:
I used to define two types separately.
And i intended to point out the core difference between `SnakeCaseCodeTokenResponse` and `SnakeCaseRefreshTokenTokenResponse` here.

I'm good with both.

Reply:
Define two types separately. PTAL